watchers/configmap_watcher_test.go

```diff
@@ -0,0 +1,153 @@
+package watchers_test
+
+import (
+	"time"
+
+	"github.com/go-logr/logr"
+	gomock "github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/project-flotta/flotta-operator/watchers"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+var _ = Describe("ConfigMap watcher", func() {
+	var (
+		configMapMock  *watchers.MockConfigMapInterface
+		watcherMock    *watchers.MockInterface
+		resultCh       chan watch.Event
+		logger         logr.Logger
+		exitCallsCount int
+	)
+
+	const (
+		configMapName          = "configMap"
+		configMapField         = "configMapField"
+		initialFieldValue      = "value1"
+		initialResourceVersion = "1"
+	)
+
+	BeforeEach(func() {
+		mockCtrl := gomock.NewController(GinkgoT())
+		configMapMock = watchers.NewMockConfigMapInterface(mockCtrl)
+		watcherMock = watchers.NewMockInterface(mockCtrl)
+		resultCh = make(chan watch.Event)
+		logger = zap.New()
+		exitCallsCount = 0
+
+		configMapMock.EXPECT().Watch(gomock.Any(), gomock.Any()).Return(watcherMock, nil).AnyTimes()
+		watcherMock.EXPECT().ResultChan().Return(resultCh).AnyTimes()
+		watcherMock.EXPECT().Stop().AnyTimes()
+	})
+
+	It("do nothing when initialFieldValue is not modified following an Add event", func() {
+		go func() {
```

Comment:
Even though I do not really liked it, I would keep the way to do test like // given // when //then. comments to keep is consistent across project.